File: Java/Programs/Shirt.java
/**
 * Shirt, granchild Product
 */
class Shirt extends Clothing {
    // atribut private
    private String color;
    private String sleeve;
    // construct kosong
    Shirt(){
    }
    // construct dengan value
    Shirt(String id, String name, String brands, double price,String size, String material, String gender, String color, String sleeve){
        super(id,name,brands,price,size,material,gender); // mengisi variabel yang dipunya orang tuanya
        this.color = color;
        this.sleeve = sleeve;
    }

    // set void
    void set_color(String color){
        this.color = color;
    }
    void set_sleeve(String sleeve){
        this.sleeve = sleeve;
    }

    // get method
    String get_color(String color){
        return this.color;
    }
    String get_sleeve(String sleeve){
        return this.sleeve;
    }

    // method untuk print list data
    void displayList(){
        System.out.println( "Product Id : "+ this.id);
        System.out.println( " ~ Name : "+ this.name);
        System.out.println( " ~ Brand : "+ this.brands);
        System.out.println( " ~ Price : "+ String.format("Rp. %,.2f", this.price));
        System.out.println( " ~ Size : "+ this.size);
        System.out.println( " ~ Material : "+ this.material);
        System.out.println( " ~ Gender : "+ this.gender);
        System.out.println( " ~ Color : "+ this.color);
        System.out.println( " ~ Sleeve : "+ this.sleeve);
    }
}

File: Python/Programs/Shirt.py
# Import kelas dasar Clothing
from Clothing import Clothing

# Derived class representing a shirt, inheriting from the Clothing class
class Shirt(Clothing):
    # Constructor
    def __init__(self, id="", name="", brands="", price=0, size="", material="", gender="", color="", sleeve=""):
        # Call the constructor of the base class (Clothing)
        super().__init__(id, name, brands, price, size, material, gender)
        # Initialize additional attributes with default or provided values
        self.color = color
        self.sleeve = sleeve

    # Setter methods

    # Set the shirt color
    def set_color(self, color):
        self.color = color

    # Set the shirt sleeve type
    def set_sleeve(self, sleeve):
        self.sleeve = sleeve

    # Getter methods

    # Get the shirt color
    def get_color(self):
        return self.color

    # Get the shirt sleeve type
    def get_sleeve(self):
        return self.sleeve

    # Method for displaying list data in a dynamic table
    def displayList(self, index):
        # Display the shirt information in a formatted table
        print(f" | {index:4} | {self.get_id():15} | {self.get_name():20} | {self.get_brands():15} | "
              f"Rp. {self.get_price():20,.2f} | {self.get_size():10} | {self.get_material():15} | "
              f"{self.get_gender():10} | {self.color:15} | {self.sleeve:15} |")
        print(" +----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+")


    # Destructor (not explicitly needed in Python, handled automatically)
